py3-gpep517.yaml

```diff
@@ -26,14 +33,33 @@ pipeline:
       uri: https://github.com/mgorny/gpep517/archive/v${{package.version}}/gpep517-v${{package.version}}.tar.gz
       expected-sha256: 965ff7cb2217fa03091face01fcb9a2b0205f58be80c8be9ccf832eab7a55414
 
-  - runs: |
-      _site_pkgs="$(python3 -c 'import site; print(site.getsitepackages()[0])')"
-      _py3ver=$(python3 -c 'import sys; print("{}.{}".format(sys.version_info.major, sys.version_info.minor))')
-
-      mkdir -p "${{targets.destdir}}"/"$_site_pkgs"
-      cp -a gpep517 "${{targets.destdir}}"/$_site_pkgs/
+subpackages:
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}
+    description: Easily download, build, install, upgrade, and uninstall Python packages
+    pipeline:
+      - runs: |
+          sitedir=usr/lib/python${{range.key}}/site-packages/
+          mkdir -p "${{targets.subpkgdir}}/$sitedir"
+          cp -a gpep517 "${{targets.subpkgdir}}/$sitedir"
+          python${{range.key}} -m compileall --invalidation-mode=unchecked-hash -r100 \
+              ${{targets.subpkgdir}}/usr/lib/
+      - uses: strip
+    dependencies:
+      provider-priority: ${{range.value}}
+      provides:
+        - ${{package.name}}
+      runtime:
+        - py${{range.key}}-tomli
+        - py${{range.key}}-installer
```

Comment:
@smoser  Are these runtime dependencies correct? I guess we can teach melange to auto-generate these in the future right?

Reply:
The dependencies are correct, but I don't like this pipeline, as dist-info is missing.